### PR TITLE
🌱 Minor fixes for CC+MP implementation

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
Follow-up to #9016

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of https://github.com/kubernetes-sigs/cluster-api/issues/5991

### CODE REVIEW

**Review by chrischdi — COMMENTED** [review]
Wow, contains some important fixes!

Thanks

/lgtm

**Comment by sbueringer** [comment]
Should be ready for review now (@chrischdi sorry I did another iteration)

/assign @killianmuldoon @chrischdi

**Comment by sbueringer** [comment]
/assign @willie-yao 
PTAL as well

File: internal/topology/check/compatibility.go [hunk 1]
```diff
@@ -196,7 +196,8 @@ func LocalObjectTemplateIsValid(template *clusterv1.LocalObjectTemplate, namespa
 // 1) InfrastructureCluster Templates are compatible.
 // 2) ControlPlane Templates are compatible.
 // 3) ControlPlane InfrastructureMachineTemplates are compatible.
-// 4) MachineDeploymentClasses have not been deleted and are compatible.
+// 4) MachineDeploymentClasses are compatible.
+// 4) MachinePoolClasses are compatible.
```
Comment (willie-yao) [inline internal/topology/check/compatibility.go:200]:
```suggestion
// 5) MachinePoolClasses are compatible.
```

**Review by willie-yao — COMMENTED** [review]
Just one minor nit from my end. Thanks for all the fixes! I definitely missed a lot of things 😅

**Comment by sbueringer** [comment]
> Just one minor nit from my end. Thanks for all the fixes! I definitely missed a lot of things 😅

No worries. I'm working since 1-2 years on ClusterClass and also missed them in the first few rounds of reviews

**Comment by chrischdi** [comment]
Only willie's nit, as of that lgtm :-)

File: internal/topology/check/compatibility.go (hunk 1 above)
Comment (sbueringer) [reply internal/topology/check/compatibility.go]: Thx, fixed!

**Review by killianmuldoon — COMMENTED** [review]
Looks good!
/lgtm
/approve

/hold
For the full CI run.

**Pull request merged**